### PR TITLE
feat : add static props to render data

### PR DESCRIPTION
**Pull request by kareraolivier**
## Static rendering

### CODE REVIEW

File: src/pages/api/get-output-json.ts [hunk 1]
```diff
@@ -0,0 +1,19 @@
+import { NextApiRequest, NextApiResponse } from "next";
+import { generateOutputJson } from "../../../generate-output-json";
+
+export default async function handler(
+  req: NextApiRequest,
+  res: NextApiResponse
+) {
+  if (req.method === "GET") {
```
Comment (NNesta) [inline src/pages/api/get-output-json.ts:8]: Can you use early returns

File: src/pages/basic-resources.tsx [hunk 2]
```diff
@@ -1,11 +1,35 @@
 import React from "react";
-import resources from "@/data/resources/basic-resources";
+import Resources from "@/data/resources/basic-resources";
```
Comment (NNesta) [inline src/pages/basic-resources.tsx:2]: can you name this function in camelcase

**Review by NNesta — CHANGES_REQUESTED** [review]

File: src/pages/api/get-output-json.ts (hunk 1 above)
Comment (kareraolivier) [reply src/pages/api/get-output-json.ts:8]: fixed

File: .gitignore [hunk 3]
```diff
@@ -26,7 +26,7 @@ yarn-debug.log*
 yarn-error.log*
 
 # local env files
-.env*.local
+.env.local
```
[the remark this answers is not in the document]
Comment (kareraolivier) [reply .gitignore:30]: It was having problems even gitignore was not ignoring it, I found that the nextjs update recommend me to use .env.local

File: src/pages/basic-resources.tsx (hunk 2 above)
Comment (kareraolivier) [reply src/pages/basic-resources.tsx:2]: fixed

File: src/utils/generate-output-json.ts [hunk 4]
```diff
@@ -0,0 +1,81 @@
...
+  Object.keys(groupedList).forEach((key) => {
+    output[key] = groupedList[key].map((item: any) => item.value);
+  });
+  return output;
+}
+
+function cleanObject(obj: any): any {
+  if (Array.isArray(obj)) {
+    return obj.map(cleanObject).filter((v) => v !== undefined && v !== null);
+  } else if (obj !== null && typeof obj === "object") {
```
Comment (NNesta) [inline src/utils/generate-output-json.ts:36]: use early return

File: src/utils/generate-output-json.ts [hunk 5]
```diff
@@ -0,0 +1,81 @@
...
+    return Object.entries(obj).reduce((acc: any, [key, value]) => {
+      if (value === undefined) {
```
Comment (NNesta) [inline src/utils/generate-output-json.ts:38]: use early return also here to make it easily understandable

**Review by NNesta — CHANGES_REQUESTED** [review]

File: src/components/Footer.tsx [hunk 6]
```diff
@@ -1,13 +1,12 @@
 import React from "react";
 import Star from "./icons/Star";
-import { filteredResources } from "../data/data-resources";
 import { getColorByAccess } from "@/utils/nodes";
-
+const footerData = ["Open Source", "Institutional Access", "Paid Service"];
```
Comment (NNesta) [inline src/components/Footer.tsx:4]: Can you call this something else like "access" and put it data folder

**Review by NNesta — CHANGES_REQUESTED** [review]

File: src/components/Footer.tsx (hunk 6 above)
Comment (AdelineA) [reply src/components/Footer.tsx:4]: Fixed

File: src/utils/generate-output-json.ts (hunk 5 above)
Comment (AdelineA) [reply src/utils/generate-output-json.ts:38]: Fixed

File: src/utils/generate-output-json.ts (hunk 4 above)
Comment (AdelineA) [reply src/utils/generate-output-json.ts:36]: Fixed

**Review by NNesta — CHANGES_REQUESTED** [review]

File: src/components/header/PracticeSection.tsx [hunk 7]
```diff
@@ -83,3 +82,5 @@ const PracticeSection = ({ title }: { title: string }) => {
 };
 
 export default PracticeSection;
+
+
```
Comment (NNesta) [inline src/components/header/PracticeSection.tsx:86]: remove the empty lines

**Review by NNesta — CHANGES_REQUESTED** [review]

File: src/data/case-studies/technologies.ts [hunk 8]
```diff
@@ -1,8 +1,11 @@
-import output from "../../../public/data.json";
+import { generateDataJson } from "@/utils/generate-output-json";
 
-const rawTechnlogies = output
-  .map((data) => data?.PARSED_MANUAL_TAGS["CASE STUDY TECH"])
-  .flat()
-  .filter((theme) => theme) as string[];
+export async function getTechnlogies() {
```
[the remark this answers is not in the document]
Comment (AdelineA) [reply src/data/case-studies/technologies.ts:3]: Fixed

File: src/data/case-studies/technologies.ts [hunk 9]
```diff
@@ -1,8 +1,11 @@
...
+  const output = await generateDataJson();
+  const rawTechnlogies = output
```
[the remark this answers is not in the document]
Comment (AdelineA) [reply src/data/case-studies/technologies.ts:5]: Fixed

File: src/components/header/PracticeSection.tsx (hunk 7 above)
Comment (AdelineA) [reply src/components/header/PracticeSection.tsx:86]: Fixed

**Review by NNesta — APPROVED** [review]
LGTM

**Pull request closed without merging**